applications/smart-distancing/libs/core.py

```diff
@@ -188,3 +197,90 @@ def non_max_suppression_fast(object_list, overlapThresh):
                                                   np.where(overlap > overlapThresh)[0])))
         updated_object_list = [j for i, j in enumerate(object_list) if i in pick]
         return updated_object_list
+
+
+    def calculate_distance_of_two_points_of_boxes(self,first_point, second_point):
+    
+        """
+        This function calculates a distance L for two input corresponding points of two detected bounding boxes.
+        it is assumed that each person is H = 170 cm tall in real scene to map the distances in the image (in pixels) to 
+        physical distance measures (in meters). 
+
+        params:
+        first_point: (x, y, h)-tuple, where x,y is the location of a point (center or each of 4 corners of a bounding box)
+        and h is the height of the bounding box. 
+        second_point: same tuple as first_point for the corresponding point of other box 
+
+        returns:
+        L:  Estimated physical distance (in centimeters) between first_point and second_point.
+
+
+        """
+
+        # estimate corresponding points distance
+        [xc1, yc1, h1] = first_point
+        [xc2, yc2, h2] = second_point
+        
+        Dx = xc2 - xc1
+        Dy = yc2 - yc1
+        
+        Lx = Dx * 170 * (1/h1 + 1/h2)/2
+        Ly = Dy * 170 * (1/h1 + 1/h2)/2
+        
+        L=math.sqrt(Lx**2+Ly**2)
+        
+        return L 
+
+
+    def calculate_box_distances(self, nn_out):
```

Comment:
Maybe it is better to implement this part with OOP.
Like detectors and loggers you should have a class that builds different distance calculation method based on config file and create a different class for each distance method with similar interfaces.
This will help you and other contributes to add more distance calculation methods easily in the future.

Reply:
Thank you for this suggestion, since it is fundamental structure modification and both applied techniques have a same basic, I will add an issue for it to implement in that better way asap.